students/YingGuo/session03/mailroom.py

```diff
@@ -0,0 +1,65 @@
+#hw3, mailroom
+
+data = {"Tom":[100,200], "Tim":[3000], "Ken": [888,888,888], "Ted":[8000,5000,500], "Jon":[1000,6000]}
+
+#Send a thank you function 
+def thank_you(data):
+	while True:
+		user_input = input("please enter the name of donor. You can enter list to see the exisitng list of donors. Or enter quit to exist current function")
+		if user_input.lower() == "list":
+			print(data.keys())
+
+		if user_input.lower() == "quit":
+			break
+		
+		elif user_input.capitalize() in list(data.keys()):
+			amount1 = int(input("how much would you like to donat?"))
+			data[user_input.capitalize()].append(amount1)
+			print("Thank you {} for your donation {}!".format(user_input.capitalize(), amount1))
+			return data
+			break
+			
+		else:
+			amount2 = int(input("how much would you like to donat?"))
+			data[user_input.capitalize()] = list()
+			data[user_input.capitalize()].append(amount2)
+			print("Thank you {} for your donation {}!".format(user_input.capitalize(), amount2))
+			return data
+			break
```

Comment:
Notice that your lines 16-20 and lines 23-28 are almost identical, except you have to add line 24. Python offers a nice way to avoid this kind of code duplication using `.setdefault()` on a dict object. We'll see it together in this week's class. Just wanted you to notice this and wanted to let you know that we should always try to avoid duplicate code as much as possible (DRY--Don't Repeat Yourself).